src/components/typeahead/typeahead.component.ts

```diff
@@ -215,6 +255,20 @@ export class TypeaheadComponent<T = any> implements OnChanges, OnDestroy {
 
     optionClickHandler(_event: MouseEvent, option: TypeaheadVisibleOption<T>): void {
         this.select(option, 'mouse');
+
+        if (Array.isArray(this.recentOptions)) {
+            // If the option is already in the array, remove it first.
+            let index = this.recentOptions.indexOf(option.value);

```

Comment:
Can we use `const` here instead of `let` as they are never reassigned.

Reply:
Done